src/hipscat_import/catalog/resume_plan.py

```diff
@@ -107,36 +108,43 @@ def read_histogram(self, healpix_order):
         - Try to find a combined histogram
         - Otherwise, combine histograms from partials
         - Otherwise, return an empty histogram
-
         """
-        full_histogram = pixel_math.empty_histogram(healpix_order)
-
-        ## Look for the single combined histogram file.
         file_name = file_io.append_paths_to_pointer(self.tmp_path, self.HISTOGRAM_BINARY_FILE)
         if file_io.does_file_or_directory_exist(file_name):
+            # Look for the single combined histogram file
             with open(file_name, "rb") as file_handle:
-                return frombuffer(file_handle.read(), dtype=np.int64)
+                full_histogram = frombuffer(file_handle.read(), dtype=np.int64)
+        else:
+            # Read the histogram from partial binaries
+            full_histogram = self.read_histogram_from_partials(healpix_order)
+        if len(full_histogram) != hp.order2npix(healpix_order):
+            raise ValueError(
+                "The histogram from the previous execution is incompatible "
+                + "with the current healpix order. To run with a different "
+                + "configuration set `resume` to False"
+            )
+        return full_histogram
 
-        ## Otherwise:
-        # - read all the partial histograms
-        # - combine into a single histogram
-        # - write out as a single histogram for future reads
-        # - remove all partial histograms
+    def _get_partial_filenames(self):
         remaining_map_files = self.get_remaining_map_keys()
         if len(remaining_map_files) > 0:
             raise RuntimeError(f"{len(remaining_map_files)} map stages did not complete successfully.")
         histogram_files = file_io.find_files_matching_path(self.tmp_path, self.HISTOGRAMS_DIR, "**.binary")
-        for file_name in histogram_files:
+        return histogram_files
+
+    def read_histogram_from_partials(self, healpix_order):
+        """Combines the histogram partials to get the full histogram."""
+        histogram_files = self._get_partial_filenames()
+        if len(histogram_files) == 0:
+            return pixel_math.empty_histogram(healpix_order)
+        for index, file_name in enumerate(histogram_files):
             with open(file_name, "rb") as file_handle:
-                full_histogram = np.add(full_histogram, frombuffer(file_handle.read(), dtype=np.int64))
-
-        file_name = file_io.append_paths_to_pointer(self.tmp_path, self.HISTOGRAM_BINARY_FILE)
-        with open(file_name, "wb+") as file_handle:
-            file_handle.write(full_histogram.data)
-        file_io.remove_directory(
-            file_io.append_paths_to_pointer(self.tmp_path, self.HISTOGRAMS_DIR),
-            ignore_errors=True,
-        )
+                full_histogram = (
```

Comment:
Personally, it would be easier to read something like:

```
full_histogram = empty_histogram(size=order)
for file in files:
  with open(blah):
    full_histogram += partial_histogram
```
Is the extra initialization step expensive?

Reply:
I adjusted the code and also added an additional check to make sure the histogram partials have consistent sizes!